### ISSUE TITLE
[device_info_plus] systemFeatures: Null is not type string

### ISSUE THREAD

**Issue by mrgnhnt96** [fluttercommunity/plus_plugins#210]
## System info

Issue occurs on: Android
Plugin name: device_info_plus
Plugin version: ^1.0.0
Flutter doctor output:

<details>
<summary>Flutter Doctor</summary>

```
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-x64, locale en-US)
    • Flutter version 2.0.4 at /Users/morganhunt/Development/versions/stable
    • Framework revision b1395592de (2 weeks ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/morganhunt/Library/Android/sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • SM G955F (mobile) • ce0317139c1ac8cf0d • android-arm64  • Android 7.0 (API 24)
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 90.0.4430.72
    ! Error: iPhone is not connected. Xcode will continue when iPhone is connected. (code -13)

• No issues found!
```
</details>

## Steps to Reproduce

I am creating an instance of `DeviceInfoPlugin`, and then calling ``.androidInfo``. 

<details>
<summary>code</summary>

```dart
abstract class DeviceInfo {
  static final _deviceInfo = DeviceInfoPlugin();

  static late _DeviceData _data;

  static Future<void> initialize() async {
   // There error happens here
    _data = await _deviceInfo.androidInfo;
  }
}
```
</details>

The issue is found in `/android_device_info.dart`. When `systemFeatures: _fromList(map['systemFeatures'] ?? []),`  (line 137) tries to evaluate, an error occurs (line 144).

## Logs

<details>
<summary>Error log</summary>

```
E/flutter ( 6173): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: type 'Null' is not a subtype of type 'String'
E/flutter ( 6173): #0      new List.from (dart:core-patch/array_patch.dart:40:5)
E/flutter ( 6173): #1      AndroidDeviceInfo._fromList
package:device_info_plus_platform_interface/model/android_device_info.dart:144
E/flutter ( 6173): #2      AndroidDeviceInfo.fromMap
package:device_info_plus_platform_interface/model/android_device_info.dart:137
E/flutter ( 6173): #3      MethodChannelDeviceInfo.androidInfo
package:device_info_plus_platform_interface/method_channel/method_channel_device_info.dart:19
E/flutter ( 6173): <asynchronous suspension>
E/flutter ( 6173): #4      DeviceInfoPlugin.androidInfo
package:device_info_plus/device_info_plus.dart:65
E/flutter ( 6173): <asynchronous suspension>
E/flutter ( 6173): #5      DeviceInfo.initialize
package:quickyy_mobile/…/device_info/device_info.dart:24
E/flutter ( 6173): <asynchronous suspension>
E/flutter ( 6173): #6      main
package:quickyy_mobile/main.dart:45
E/flutter ( 6173): <asynchronous suspension>
E/flutter ( 6173):

```
</details>

<details>
<summary>Data trying to evaluate</summary>

This should be a list of only Strings

```
[0]:"android.hardware.sensor.proximity"
[1]:"com.samsung.android.sdk.camera.processor"
[2]:"com.samsung.feature.aodservice_v03"
[3]:"com.sec.feature.motionrecognition_service"
[4]:"com.sec.feature.cover.sview"
[5]:"com.samsung.android.camera.iris"
[6]:"android.hardware.sensor.accelerometer"
[7]:"android.hardware.faketouch"
[8]:"com.sec.feature.multiwindow.ensure_docked_view"
[9]:"com.samsung.feature.virtualscreen"
[10]:"com.samsung.feature.device_category_phone_high_end"
[11]:"android.hardware.usb.accessory"
[12]:"com.samsung.feature.hdr_capable"
[13]:"android.software.backup"
[14]:"android.hardware.touchscreen"
[15]:"android.hardware.touchscreen.multitouch"
[16]:"android.software.print"
[17]:"com.samsung.feature.mirrorlink_fw"
[18]:"android.hardware.sensor.heartrate"
[19]:"android.software.voice_recognizers"
[20]:"android.hardware.fingerprint"
[21]:"com.samsung.android.knox.knoxsdk"
[22]:"android.hardware.sensor.gyroscope"
[23]:"com.sec.feature.slocation"
[24]:"com.samsung.android.sdk.camera.processor.haze"
[25]:"com.samsung.android.sdk.camera.ipx"
[26]:"android.hardware.opengles.aep"
[27]:"com.sec.feature.fingerprint_manager_service"
[28]:"com.sec.feature.sensorhub"
[29]:"com.sec.feature.overlaymagnifier"
[30]:"android.hardware.bluetooth"
[31]:"android.hardware.camera.autofocus"
[32]:"android.hardware.telephony.gsm"
[33]:"com.sec.feature.cocktailpanel"
[34]:"com.samsung.android.bio.face"
[35]:"android.software.sip.voip"
[36]:"android.hardware.vr.high_performance"
[37]:"com.samsung.feature.samsung_experience_mobile"
[38]:"android.hardware.usb.host"
[39]:"android.hardware.audio.output"
[40]:"android.software.verified_boot"
[41]:"android.hardware.camera.flash"
[42]:"android.hardware.camera.front"
[43]:"android.hardware.sensor.hifi_sensors"
[44]:"com.sec.feature.multiwindow.snap_view"
[45]:"com.samsung.feature.hmt"
[46]:"android.hardware.screen.portrait"
[47]:"android.hardware.nfc"
[48]:"com.sec.android.mdm"
[49]:"com.sec.feature.cover.clearcover"
[50]:"android.hardware.sensor.stepdetector"
[51]:"android.software.home_screen"
[52]:"com.sec.feature.barcode_emulator"
[53]:"com.samsung.android.sdk.camera.processor.panorama"
[54]:"android.hardware.microphone"
[55]:"com.samsung.android.sdk.camera.processor.effect"
[56]:"android.software.vr.mode"
[57]:"android.hardware.bluetooth_le"
[58]:"com.sec.feature.edge_v03"
[59]:"android.hardware.sensor.compass"
[60]:"android.hardware.touchscreen.multitouch.jazzhand"
[61]:"android.hardware.sensor.barometer"
[62]:"android.software.app_widgets"
[63]:"android.software.input_methods"
[64]:"android.hardware.sensor.light"
[65]:"android.hardware.vulkan.version"
[66]:"com.samsung.feature.galaxyfinder_v7"
[67]:"android.software.device_admin"
[68]:"android.hardware.camera"
[69]:"android.hardware.screen.landscape"
[70]:"com.samsung.android.authfw"
[71]:"com.samsung.android.api.version.2402"
[72]:"com.samsung.android.api.version.2403"
[73]:"com.sec.feature.cover"
[74]:"com.sec.feature.findo"
[75]:"android.software.managed_users"
[76]:"com.sec.feature.nsflp"
[77]:"android.software.webview"
[78]:"com.samsung.feature.radiobasedlocation"
[79]:"android.hardware.sensor.stepcounter"
[80]:"android.hardware.camera.capability.manual_post_processing"
[81]:"com.sec.feature.spo2"
[82]:"android.hardware.camera.any"
[83]:"com.samsung.android.sdk.camera.processor.dof"
[84]:"com.samsung.android.sdk.camera.processor.gif"
[85]:"com.samsung.android.sdk.camera.processor.hdr"
[86]:"com.samsung.android.sdk.camera.processor.lls"
[87]:"android.hardware.camera.capability.raw"
[88]:"android.software.connectionservice"
[89]:"android.hardware.touchscreen.multitouch.distinct"
[90]:"android.hardware.location.network"
[91]:"com.sec.android.secimaging"
[92]:"android.software.sip"
[93]:"android.hardware.camera.capability.manual_sensor"
[94]:"com.samsung.feature.mmfw_monitor"
[95]:"android.hardware.camera.level.full"
[96]:"android.hardware.wifi.direct"
[97]:"android.software.live_wallpaper"
[98]:"com.sec.feature.multiwindow.long_device_snap_mode"
[99]:"android.software.freeform_window_management"
[100]:"android.hardware.nfc.hcef"
[101]:"android.hardware.location.gps"
[102]:"com.sec.feature.fingerprint_ui"
[103]:"android.software.midi"
[104]:"android.hardware.nfc.hce"
[105]:"com.sec.feature.cover.nfc_authentication"
[106]:"com.sec.feature.support_mst"
[107]:"android.hardware.wifi"
[108]:"android.hardware.location"
[109]:"android.hardware.vulkan.level"
[110]:"com.sec.feature.people_edge_notification"
[111]:"com.sec.feature.cover.flip"
[112]:"com.sec.feature.cover.nfcledcover"
[113]:"android.hardware.telephony"
[114]:"com.sec.android.smartface.smart_stay"
[115]:null
```
</details>

Flutter Analyze
```
No issues found! (ran in 6.2s)
```

**Comment by miquelbeltran** [comment]
Hi @mrgnhnt96 there was a similar issue fixed some days ago: https://github.com/fluttercommunity/plus_plugins/pull/191 and the corresponding issue https://github.com/fluttercommunity/plus_plugins/issues/184 is this related to those?

**Comment by mrgnhnt96** [comment]
> Hi @mrgnhnt96 there was a similar issue fixed some days ago: #191 and the corresponding issue #184 is this related to those?

Yes, that is the same issue. I saw the fix to run `flutter pub upgrade` and it worked, thanks!
